Composer/packages/client/src/components/QnA/CreateQnAModal.tsx

```diff
@@ -0,0 +1,732 @@
+              getKey={(item) => item.name}
+              items={sortedKbs}
+              layoutMode={DetailsListLayoutMode.justified}
+              selection={selectedKB}
+              selectionMode={SelectionMode.single}
+              onRenderRow={onRenderRow}
+            />
+            {kbLoading && <Spinner label={kbLoading} labelPosition="bottom" />}
+            {sortedKbs.length === 0 && !kbLoading && (
+              <p style={{ textAlign: 'center' }}> {formatMessage('No avaliable knowledge base')} </p>
+            )}
+          </div>
+        </div>
+        <DialogFooter>
+          {loading && <Spinner label={loading} labelPosition="right" styles={{ root: { float: 'left' } }} />}
+          <DefaultButton disabled={!!loading} text={formatMessage('Back')} onClick={() => setCurrentStep('resource')} />
+          <PrimaryButton
+            disabled={!!loading || (!userProvidedTokens && !tenantId) || !subscriptionId || !selectedKb}
+            text={formatMessage('Next')}
+            onClick={onSubmitImportKB}
+          />
+          <DefaultButton disabled={!!loading} text={formatMessage('Cancel')} onClick={props.onDismiss} />
+        </DialogFooter>
+      </div>
+    );
+  };
+
+  const renderCurrentStep = () => {
+    switch (currentStep) {
+      case 'intro':
+        return renderIntroStep();
+      case 'resource': {
+        if (nextAction === 'portal') {
+          return renderChooseResourceStep();
+        }
+        break;
+      }
+      case 'knowledge-base':
+        return renderKnowledgeBaseSelectionStep();
+      default:
+        return null;
+    }
+  };
+
+  useEffect(() => {
+    switch (currentStep) {
+      case 'intro':
+        setDialogTitle(formatMessage('Add QnA Maker knowledge base'));
+        break;
+      case 'resource':
+        if (nextAction === 'portal') {
+          setDialogTitle(formatMessage('Select source knowledge base location'));
+        }
+        break;
+      case 'knowledge-base':
+        setSelectedKb(undefined);
+        setDialogTitle(formatMessage('Choose a knowledge base to import'));
+        break;
+    }
+  }, [currentStep]);
+
+  const handleDismiss = () => {
+    onDismiss?.();
+    setInitialName('');
+    actions.createQnADialogCancel({ projectId });
+    TelemetryClient.track('AddNewKnowledgeBaseCanceled');
+  };
+
+  const onFormDataChange = (data, disabled) => {
+    setFormData(data);
+    setDisabled(disabled);
+  };
+
+  const onSubmitFormData = () => {
+    if (disabled || !formData) {
+      return;
+    }
+    onSubmit(formData);
+    setInitialName('');
+    TelemetryClient.track('AddNewKnowledgeBaseCompleted', { source: formData.urls?.length ? 'url' : 'none' });
+  };
+
+  const onSubmitImportKB = async () => {
+    if (key && token && selectedKb && formData) {
+      // TODO: add to all matched language or ask user for specific locale.
+      const createdOnLocales = locales.filter((item) => localeToLanguage(item) === selectedKb.language);
```

Comment:
Update to auto match locales according to language of selected KB. @srinaath @boydc2014